src/app/core/user/user-details/user-details.component.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  Component,
+  computed,
+  effect,
+  inject,
+  input,
+  output,
+  ChangeDetectionStrategy,
+} from "@angular/core";
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from "@angular/forms";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatInputModule } from "@angular/material/input";
+import { MatSelectModule } from "@angular/material/select";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { MatButtonModule } from "@angular/material/button";
+import { Role, UserAccount } from "../user-admin-service/user-account";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { MatSlideToggleModule } from "@angular/material/slide-toggle";
+
+/**
+ * Reusable user account details form component.
+ * Used by EntityUserComponent and ProfileComponent for consistent user management UI.
+ */
+@UntilDestroy()
+@Component({
+  selector: "app-user-details",
+  templateUrl: "./user-details.component.html",
+  styleUrls: ["./user-details.component.scss"],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    ReactiveFormsModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatSelectModule,
+    MatTooltipModule,
+    MatButtonModule,
+    MatSlideToggleModule,
+  ],
+})
+export class UserDetailsComponent {
+  private fb = inject(FormBuilder);
+
+  userAccount = input<UserAccount | null>();
+  availableRoles = input<Role[]>([]);
```

Comment:
who is responsible for loading this? I think maybe this is better handled by this component itself internally to load available roles?